tactics/hints.ml

```diff
@@ -1101,8 +1101,17 @@ type db_obj = {
   db_ts : TransparentState.t;
 }
 
-let cache_db {db_name=name; db_use_dn=b; db_ts=ts} =
-  searchtable_add (name, Hint_db.empty ~name ts b)
+let warn_mismatch_create_hintdb = CWarnings.create ~name:"mismatched-hint-db" ~category:CWarnings.CoreCategories.automation
+    Pp.(fun {db_name;db_use_dn} ->
+        str "Hint Db " ++ str db_name ++ str " already exists and " ++
+        (if db_use_dn then str "is not" else str "is") ++ str " discriminated.")
```

Comment:
```suggestion
        str "Hint Db " ++ str db_name ++ str " already exists and is " ++
        (if db_use_dn then str "not" else mt ()) ++ str " discriminated.")
```

`if db_use_dn then str "not" else` - reads a bit strangely because `db_use_dn` normally means discriminated but it maps to "not discriminated".  Or is actually incorrect?

Reply:
db_use_dn is what was requested, so the opposite is what already exists.